src/ORM/Filters/ComparisonFilter.php

```diff
@@ -32,6 +33,44 @@ abstract protected function getOperator();
      */
     abstract protected function getInverseOperator();
 
+    public function matches(mixed $toMatch): bool
+    {
+        $negated = in_array('not', $this->getModifiers());
+
+        // can't just cast to array, because that will convert null into an empty array
+        $values = $this->getValue();
+        if (!is_array($values)) {
+            $values = [$values];
+        }
+
+        // This is essentially a in_array($toMatch, $values) check, with some special handling.
+        $hasMatch = false;
+        foreach ($values as $value) {
+            $doesMatch = $this->match($toMatch, $value);
+
+            // Any match is a match
+            if ($doesMatch) {
+                $hasMatch = true;
+                break;
+            }
+        }
+
+        // Respect "not" modifier.
+        if ($negated) {
+            $hasMatch = !$hasMatch;
+        }
+
+        return $hasMatch;
+    }
+
+    protected function match(mixed $objectValue, mixed $filterValue): bool
+    {
+        // We can't add an abstract method but we want to enforce the method signature for any subclasses
```

Comment:
Why don't we just use an abstract protected method? https://www.php.net/manual/en/language.oop5.abstract.php

Reply:
I'm not sure what specific information you want me to get from the docs... but in any case we can't use abstract because that will mean any custom subclasses of `ComparisonFilter` will need to implement a new method, which makes it a breaking change.

Reply:
Ah ok makes sense, just update the inline comment to add `we can't use abstract because that will mean any custom subclasses of ComparisonFilter would need to implement this method, which makes it a breaking change.`

We could raise a card here to changes this (and the other example of this) to abstract methods for CMS 6

Reply:
Comment amended to include that context.
Issue created: https://github.com/silverstripe/silverstripe-framework/issues/10933